k2/csrc/fsa_algo.h

```diff
@@ -659,6 +660,41 @@ void InvertHost(FsaOrVec &src, Ragged<int32_t> &src_aux_labels, FsaOrVec *dest,
 FsaOrVec RemoveEpsilonSelfLoops(FsaOrVec &src,
                                 Array1<int32_t> *arc_map = nullptr);
 
+
+/*
+  Replace, in `index`, `labels symbol_range_begin <= label < symbol_range_begin+src.Dim0()`
+  with the Fsa indexed `label - symbol_range_begin` in `src`, identifying the
+  source and destination states of the arc in `src` with the initial and
+  final states in `src[label - symbol_range_begin]`. Arcs with labels outside
+  this range are just copied. Labels on final-arcs in `src` (which will be -1)
+  would be set to 0(epsion) in the result.
```

Comment:
0 (epsilon)

Also please edit it so the whitespace lines up (is all on the same cursor position)
I originally wrote this in one of these github windows, and you can't line it up because
the GitHub text window uses non-fixed-width characters.